src/gbdxm.cpp

```diff
@@ -17,309 +17,210 @@
-            cerr << "Cannot write to output, " << args.gbdxFile << " is a directory.";
-            exit(1);
-        }
+    // Make sure the output file can be created
+    DG_CHECK(!fs::is_directory(args.gbdxFile), "Cannot write to output, %s is a directory.", args.gbdxFile);
 
-        auto parentPath = fs::path(args.gbdxFile).parent_path();
-        if(!parentPath.empty() && !fs::exists(parentPath)) {
-            printVerbose(args, "Creating directory " + parentPath.string() + "...");
-            fs::create_directories(parentPath);
-        }
+    auto parentPath = fs::path(args.gbdxFile).parent_path();
+    if(!parentPath.empty() && !fs::exists(parentPath)) {
+        DG_LOG(gbdxm, info) << "Creating directory " << parentPath.string();
+        fs::create_directories(parentPath);
+    }
 
-        // Check the input files
-        bool missingFiles = false;
-        if(args.labelsFile.empty() && args.metadata->labels().empty()) {
-            cerr << "Labels not given in a file or json" << endl;
-            missingFiles = true;
+    auto& package = *args.package;
+    auto& metadata = package.metadata();
 
-        } else if(!args.labelsFile.empty() && (!fs::exists(args.labelsFile) || fs::is_directory(args.labelsFile))) {
-            cerr << "Label file does not exist at '" << args.labelsFile << "'" << endl;
-            missingFiles = true;
-        }
+    stringstream ess;
 
-        // Find out the total file size while we're checking the files
-        int64_t totalFileSize = 0;
-        for(const auto& mapItem : args.modelFiles) {
-            if(!fs::exists(mapItem.second) || fs::is_directory(mapItem.second)) {
-                cerr << "File does not exist for " << mapItem.first << " at '" << mapItem.second << "'" << endl;
-                missingFiles = true;
-                continue;
-            }
-            totalFileSize += fs::file_size(mapItem.second);
-        }
-
-        args.metadata->setSize(totalFileSize);
+    // Check the input files
+    if(args.labelsFile.empty() && metadata.labels().empty()) {
+        ess << "Labels not given in a file or JSON" << endl;
+    } else if(!args.labelsFile.empty() && (!fs::exists(args.labelsFile) || fs::is_directory(args.labelsFile))) {
+        ess << "Label file does not exist at '" << args.labelsFile << "'";
+    }
 
-        if(missingFiles) {
-            exit(1);
+    // Find out the total file size while we're checking the files
+    int64_t totalFileSize = 0;
+    for(const auto& mapItem : args.modelFiles) {
+        if(!fs::exists(mapItem.second) || fs::is_directory(mapItem.second)) {
+            ess << "File does not exist for " << mapItem.first << " at '" << mapItem.second << "'" << endl;
+            continue;
         }
+        totalFileSize += fs::file_size(mapItem.second);
+    }
 
-        // Create the model
-        unique_ptr<ModelPackage> model(ModelPackage::create(*args.metadata));
+    DG_CHECK(ess.str().empty(), ess.str().c_str());
```

Comment:
Regardless of throw or log, I mean that `ess` could contain a "%".  We want to make sure that isn't used by the formatter.